COM-3573 B2B P2: Filter B2B account dropdown by Seller

rajeshdatir opened the pull request:
Filter the B2B account dropdown based on the Seller. Only B2B accounts a Seller is assigned to should be available. This applies to searching/filtering as well.

- Refactor the mozuPaginatedSearchableDropdown options to accept a single `filter` string with all static filters defined.
- Add the Seller's user ID as a default filter using the `salesrep.userid` filter.
- Add a configuration for the sort direction on the mozuPaginatedSearchableDropdown.

kibo-rajeshdatir committed:
COM-3573 B2B P2: Filter "Account search" dropdown based on seller logged in

sagarraut0007 commented:
Changes looks good to me

michaelsadlonkibo commented on scripts/widgets/mz-search-pagination-dd.js:
```diff
@@ -20,7 +20,11 @@ define(["jquery", "underscore"], function ($, _) {
+        if (options.userId) {
+            filterUserIdStr += "salesrep.userid" + ' ' + options.filterOption1 + ' ' + options.userId + " and ";
+        }
+        var dataSet = new options.model({ pageSize: options.pageSize, filter: filterUserIdStr +  options.filterKey + ' ' + options.filterOption1 + ' ' + options.filterValue, sortBy: options.textField + ' ' + 'asc' });
```
This is a generic component so we can't hardcode anything inside it. Make your changes using the config options for the component. You will need to revert all your changes in this file.

SharmaBhushan replied:
Done.

michaelsadlonkibo commented on scripts/modules/b2b-account/quotes.js:
```diff
@@ -154,7 +155,8 @@ define([
                         filterKey: 'isActive',
-                        filterValue: 'true'
+                        filterValue: 'true',
+                        userId: userId
```
We can't "hardcode" anything like this. This component needs to be generic and work for _any mozu type or any set of filters_. Find another solution that is generic.

Work with Bhushan if you have questions about how this component works. He wrote it.

SharmaBhushan replied:
Done

SharmaBhushan committed:
COM-3573- Added account search configuration field

configure sales rep user id and sort direction

Combine filter string added remove extra configuration

michaelsadlonkibo reviewed (changes requested):
I still don't think this fix matches my expectations. While the filter for `salesrep.userid` is now configurable this component is not being treated as the generic component it is. There should be no concept of a sales rep and associated filter.

Ideally we would fix this to accept a single generic `filter` parameter that specifies all the filters or a collection of filters and their values.

Let's meet separately and discuss this in more detail.

michaelsadlonkibo reviewed (approved):
Thanks for making those changes. I think it's ready to go now.

SharmaBhushan committed:
COM-3573 COM-3789 Fixed dropdown select issue
